### PR TITLE
RavenDB-14304 fix issues when appeneding values with different dimens…

### PR DESCRIPTION
**Pull request by karmeli87**
…ions

### CODE REVIEW

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 1]
```diff
@@ -1113,6 +1095,32 @@ Reader.SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
             return context.LastDatabaseChangeVector;
         }
 
+        private bool ValueTooFar(TimeSeriesSegmentHolder segmentHolder, Reader.SingleResult current)
+        {
+            var deltaInMs = (current.Timestamp.Ticks / 10_000) - segmentHolder.BaselineMilliseconds;
```
Comment (ppekrol) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:1296]: Can we wrap this in something similar to `Size`
I'm afraid that without that it is a potential place for easy bug introduction. Let's talk about this tomorrow.

**Review by ppekrol — APPROVED** [review]

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:1296]: Seems reasonable to do so, yes

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs [hunk 2]
```diff
@@ -1113,6 +1095,32 @@ Reader.SingleResult ToResult(TimeSeriesOperation.AppendOperation element)
...
+                {
+                    var fullValues = new Span<double>(fullValuesBuffer.Ptr, segmentNumberOfValues);
+                    current.Values.Span.CopyTo(fullValues);
+                    for (int i = current.Values.Length; i < fullValues.Length; i++)
+                    {
+                        fullValues[i] = double.NaN;
+                    }
+
+                    current.Values = new Memory<double>(fullValues.ToArray());
+                }
```
Comment (ayende) [inline src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: You are returning the allocation here, but still using it.
This is _not_ valid

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: `fullValues.ToArray()` copy them to managed memory?

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs:1296]: Let's do in a separate issue: https://issues.hibernatingrhinos.com/issue/RavenDB-14577

File: src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: Oh, correct, but then why bother will all the song and ance? Just allocate them once directly and save all the complexity.

Comment (karmeli87) [reply src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs]: right.

**Review by ayende — APPROVED** [review]

**Pull request merged**